caffe2/ideep/operators/quantization/int8_relu_op.cc

```diff
@@ -0,0 +1,41 @@
+#include <caffe2/ideep/ideep_utils.h>
+
+namespace caffe2 {
+
+class IDEEPInt8ReluOp final : public IDEEPOperator {
+ public:
+  USE_IDEEP_DEF_ALIASES();
+  USE_IDEEP_OPERATOR_FUNCTIONS();
+
+  IDEEPInt8ReluOp(const OperatorDef& operator_def, Workspace* ws)
+      : IDEEPOperator(operator_def, ws), alpha_(0.0) {
+    // Figure out the Relu descriptor.
+    if (operator_def.type().substr(0, 8) == "Int8Relu") {
+      alpha_ = 0.0;
+    } else {
+      LOG(FATAL) << "Unsupported Relu method: " << operator_def.type();
+    }
+  }
+  ~IDEEPInt8ReluOp() override {}
+
+  bool RunOnDevice() override {
+    const auto& X = Input(INPUT);
+    auto* Y = Output(OUTPUT);
+
+    ideep::eltwise_forward::compute(
+        X, *Y, ialgo::eltwise_relu, iprop::forward_inference, alpha_);
+
+    return true;
+  }
+
+ private:
+  ikey op_key_;
```

Comment:
What does `ikey` do?

Reply:
ikey is the hash key for ideep operators to fetch the op from ideep lru.

Reply:
@yinghai We cache the hash key in the op in order to mitigate the overhead of creating the key on each iteration. The overhead would become obvious when the workload is small running on multiple cores, e.g. single-batch inference on a single Xeon socket.